docs/lessons/music.rst

```diff
@@ -0,0 +1,129 @@
+
+Wolfgang Amadeus Microbit
++++++++++++++++++++++++++
+
+Creating your own tunes is easy!
+
+Each note has a name (like ``C#`` or ``F``), an octave (telling MicroPython if
+it should play a high, medium or low version of the note) and a duration (how
+long it lasts through time). Octaves are indicated by a number, as are
+durations. The higher the value of the duration the longer it will last. Such
+values are related to each other - for instance, a duration of ``4`` will last
+twice as long as a duration ``2`` (and so on). If you use the note name ``R``
+then MicroPython will play a rest (i.e. silence) for the specified duration.
+
+Each note is expressed as a string of characters like this::
+
+    NOTE[octave][:duration]
+
+For example, ``"A1:4"`` refers to the note named ``A`` in octave number ``1``
+to be played for a duration of ``4``.
+
+To create a melody, you simply make a list of notes (in exactly the same way
+as you would create an animation with a list of images). For example, here's
+how to make MicroPython play opening of "Frere Jaques"::
+
+    import music
+
+    tune = ["C4:4", "D4:4", "E4:4", "C4:4", "C4:4", "D4:4", "E4:4", "C4:4",
+            "E4:4", "F4:4", "G4:8", "E4:4", "F4:4", "G4:8"]
+    music.play(tune)
+
+.. note::
+
+    MicroPython can help you from typing too much. It'll remember the octave
+    and duration values until you next change them. As a result, the example
+    above can be re-written as::
+
+        import music
+
+        tune = ["C4:4", "D", "E", "C", "C", "D", "E", "C", "E", "F", "G:8",
+                "E:4", "F", "G:8"]
+        music.play(tune)
+
+    Notice how the octave and duration values only change when they have to.
+
+Sound Effects
++++++++++++++
+
+It's possble to make tones that are not musical notes. For example, here's how
+to create a Police siren effect::
+
+    import music
+
+    while True:
+        for freq in range(880, 1760, 16):
+            music.pitch(freq, 6)
+        for freq in range(1760, 880, -16):
+            music.pitch(freq, 6)
+
+
+Notice how the ``music.pitch`` *method* is used in this instance. It expects a
+frequency. For example, the frequency of ``440`` is the same as a concert ``A``
+in standard Western classical music.
+
+In the example above the ``range`` function is used to generate ranges of
+numeric values. The three arguments are the start value, end value and step
+size. Therefore, the first use of ``range`` is saying, in English, "create a
+range of values between 880 and 1760 in steps of 16". The second use of
+``range`` is saying, "create a range of values between 1760 and 880 in steps of
+-16".
+
+Because the siren should last forever it's wrapped in an infinite ``while``
+loop. Importantly, we have introduced a new sort of a loop inside the ``while``
+loop: the ``for`` loop. In English it's like saying, "for each item in some
+collection, do some thing". Specifically in the example above, it's saying,
```

Comment:
Could be both but "something" is more common.

Keep these comments coming, they're *invaluable* and really helpful. As you've probably figured out, I'm writing all this during my commute in to/out of London, lunch time and in the evenings. It's hard to tell if it's making any sense...

:-)